guineaPigFacts/src/main/java/com/factsdemo/guineaPigFacts/filters/PasswordFilter.java

```diff
@@ -0,0 +1,83 @@
+package com.factsdemo.guineaPigFacts.filters;
+
+import org.springframework.stereotype.Component;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Filter class that removes passwords from responses being sent back to the client
+ */
+@Component
+public final class PasswordFilter implements Filter {
+    /**
+     * Called before Filter is used & sets the configuration object
+     * @param filterConfig
+     */
+    @Override
+    public void init(FilterConfig filterConfig) {
+    }
+
+    /**
+     * Called with every request/response pair sent due to a client request for a resource
+     * @param request
+     * @param response
+     * @param chain
+     * @throws IOException
+     * @throws ServletException
+     */
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response,
+                         FilterChain chain) throws IOException, ServletException {
+
+        WrappedResponse wrappedResponse = new WrappedResponse((HttpServletResponse)response);
+
+        chain.doFilter(request, wrappedResponse);
+
+        String responseContent = new String(wrappedResponse.getDataStream());
+        String filteredResponse = filterPassword(responseContent);
+        byte[] responseToSend = restResponseBytes(filteredResponse);
+
+        response.getOutputStream().write(responseToSend);
+
+    }
+
+    /**
+     * Called after the filter is done
+     */
+    @Override
+    public void destroy() {
+    }
+
+    /**
+     * Searches the response that will be sent to client and removes the password if it was found
+     * @param responseContent
+     * @return String filtered response
+     */
+    private String filterPassword(String responseContent) {
+        StringBuilder filteredContent = new StringBuilder();
+        String [] parseContent = responseContent.split(",");
+        for(int i=0; i < parseContent.length; i++) {
+            if(!parseContent[i].contains("\"password\":")) {
+                if(i < parseContent.length-1) {
+                    filteredContent=filteredContent.append(parseContent[i]+",");
+                }
+                else {
+                    filteredContent=filteredContent.append(parseContent[i]);
+                }
+            }
+        }
```

Comment:
Are you parsing JSON here?  If so, a better solution would be to use the Jackson ObjectMapper.readValue method to parse the string into a User object.  You could then set the value of Password to null.

Parsing JSON is very tricky.  There are tons of edge-cases and gotchas.  Best to leave this stuff to the libraries that are well-tested.  As a bonus, Jackson is baked into Spring Boot so you already have it.  Might as well use it :)

Reply:
Some of the responses are JSON and others are just phrases i.e. "removed" and "Deleted."  The unit tests error because there is a JSON parse error when trying to parse those phrases into JSON. From what I understand, I can do 3 things:
1. Set up a try/catch with an empty catch so when a JSON parse fails nothing happens, but from what I've read empty try/catches is bad practice.
2. Set up the filter so it is only in service for the URLs that will use the filter, but that seems like it would defeat the purpose of trying to automate filtering.
3. Parsing the content as a String which is tricky from the edge-cases as you commented above.

Reply:
Actually, I think I got it. Instead of just returning nothing in the try/catch i just returned the initial response instead in the catch.

Reply:
nice.  yeah good solution.

regarding empty try/catches.  In general, I would say it's a bad practice.  There are definitely times where you might swallow an exception, but it's so rare that you'd always want to leave a comment in the empty catch so that future devs understand that the catch was left empty on purpose.